vllm/entrypoints/openai/api_server.py

```diff
@@ -629,6 +631,13 @@ async def fake_stream_generator() -> AsyncGenerator[str, None]:
                               tokenizer_mode=engine_args.tokenizer_mode,
                               trust_remote_code=engine_args.trust_remote_code)
 
+    if chat_template or tokenizer.chat_template:
```

Comment:
Huh yeah, thanks for pointing that out: https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils_base.py#L1795

These things change so quick!

Reply:
Okay, have changed this to still give a warning if undefined, and print out the template in use as in info log.
```
    if chat_template is not None:
        tokenizer.chat_template = chat_template

    if tokenizer.chat_template or tokenizer.default_chat_template:
        logger.info(
            f"Chat template:\n{tokenizer.chat_template or tokenizer.default_chat_template}")
    else:
        logger.warning(
            "No chat template loaded, the chat endpoint will be disabled.")
```

Reply:
Wdym by `will be disabled` here? it won't use any chat template right?

Now thinking more about this, I feel like the decision of formatting chat templates or not should leave it to the users.

Especially for people who just pass the tokens via `prompt_token_ids` instead of using `prompt` parameters.

Reply:
In essence, the chat completions API needs a template for formatting input. You aren't converting from a list of messages (role/content pairs)  to a formatted piece of text without *something* driving what to do. Without a template, it will error out. However, as long as you're not using an early version of Transformers that supports chat templates but not the default template, you should be fine...

Regarding the `prompt_token_ids`, they are not applicable in the chat completions API since it requires a message list consisting of strings only. Integers not allowed.

In my view, choosing to use the chat API itself is an opt-in decision. The chat API inherently needs a template to function correctly, involving some necessary internal/external configurations for how that template should work on the server side.

If users prefer client-side formatting, the regular completions API is more suitable. It allows the submission of pre-formatted `prompt` or `prompt_token_ids`, offering greater flexibility in formatting.